### PR TITLE
Include all subcommands in top-level `--help` output

### LINKED ISSUE #238: Plain `--help` should also show `diff --help`
So that a single `--help` invocation is sufficient to learn how to use the tool.

Blocked on https://github.com/clap-rs/clap/issues/1334 (unless we work around this somehow, which I would be in favor doing, unless it's very complicated).

### PR DESCRIPTION
**Pull request by Enselic**
So that users only have to issue one command to see how all subcommands are supposed to be used.

Closes #238

### CODE REVIEW

**Comment by Enselic** [comment]
Actually, let me see if I can update the help text with examples also with flatten_help...

Edit: Done.

File: cargo-public-api/src/main.rs [hunk 1]
```diff
@@ -145,8 +146,25 @@ struct DiffArgs {
...
 
-    /// What to diff. See `cargo public-api diff --help` for examples and more
-    /// info.
+    #[clap(verbatim_doc_comment)]
+    /// What to diff.
+    ///
+    /// EXAMPLES
+    /// ========
+    ///
+    /// Diffing working tree against a published version of a specific crate in the workspace:
```
Comment (Emilgardis) [inline cargo-public-api/src/main.rs]:
```suggestion
    /// Diffing current working tree against a published version of `specific-crate` in the workspace:
```

File: cargo-public-api/src/main.rs [hunk 2]
```diff
@@ -145,8 +146,25 @@ struct DiffArgs {
...
+    ///
+    ///     cargo public-api -p specific-crate diff 1.2.3
+    ///
+    /// Diffing between commits:
+    ///
+    ///     cargo public-api diff 1.2.3 cargo public-api diff v0.2.0..v0.3.0
```
Comment (Emilgardis) [inline cargo-public-api/src/main.rs]:
```suggestion
    ///     cargo public-api diff v0.2.0..v0.3.0
```

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: cargo-public-api/src/main.rs (hunk 2 above)
Comment (Enselic) [reply cargo-public-api/src/main.rs]: Oops, thanks :)

File: cargo-public-api/src/main.rs (hunk 1 above)
Comment (Enselic) [reply cargo-public-api/src/main.rs]: 👍 

Inspired by your suggestion I reworded it a bit further to make it shorter so it fits on a single 100 char line.

**Review by Emilgardis — APPROVED** [review]

**Pull request merged**